core/workspace_svg.js

```diff
@@ -1076,9 +1076,14 @@ Blockly.WorkspaceSvg.prototype.deleteVariableById = function(id) {
  * @package
  */
 Blockly.WorkspaceSvg.prototype.createVariable = function(name, opt_type, opt_id) {
+  var variableInMap = (this.getVariable(name) != null);
   var newVar = Blockly.WorkspaceSvg.superClass_.createVariable.call(this, name,
```

Comment:
Do you still need to call the superclass `createVariable` if you've already determined that the variable exists?

Reply:
Ah, good question.  I don't quite have time to test at the moment, but I originally left it in because when you create a variable through the button this gets called and I wanted to make sure we don't suppress the duplicate message when you try and create a variable by the same name.  But maybe that error gets caught in the prompt. I'll look into it.

Reply:
Well, at least the case where the user creates a variable through the create button is handled so I could skip this.

There is still at least 1 behavior change though.  Right now importing an existing variable from xml fails.  If I hide this behind the check it won't.  So we'd have to decide that we actually want that.